MantidPlot/src/Mantid/MantidUI.cpp

```diff
@@ -2043,39 +2043,38 @@ InstrumentWindow* MantidUI::getInstrumentView(const QString & wsName, int tab)
 
   //Need a new window
   const QString windowName(QString("InstrumentWindow:") + wsName);
-  InstrumentWindow *insWin = new InstrumentWindow(wsName,QString("Instrument"),appWindow(),windowName);
+  
   try
   {
-    insWin->init();
+    InstrumentWindow *insWin = new InstrumentWindow(
+        wsName, QString("Instrument"), appWindow(), windowName);
+
+    insWin->getInstrumentWidget()->selectTab(tab);
+
+    appWindow()->addMdiSubWindow(insWin);
+
+    QApplication::restoreOverrideCursor();
+    return insWin->getInstrumentWidget();
```

Comment:
Does this method really need to return the `InstrumentWidget`. As this is MantidPlot, I think we should only be working with `InstrumentWindows`.